src/index.ts

```diff
@@ -130,7 +144,22 @@ function makeSourceMapAndFinish(
   );
 
   setModuleMeta(loaderContext, instance, fileVersion);
-  callback(null, output, sourceMap);
+
+  // if we don't have an input source map, we only need to return the newly generated source map
+  // this avoids that we have to make a possibly expensive call to the source-map lib
+  if (inputSourceMap === undefined) {
+    callback(null, output, sourceMap);
+    return;
+  }
+
+  // otherwise we have to make a mapping to the input source map which is asynchronous
+  mapToInputSourceMap(sourceMap, loaderContext, inputSourceMap)
+    .then((mappedSourceMap: RawSourceMap) => {
```

Comment:
Do we need the type annotation on the method? Doesn't TypeScript infer it?

Reply:
yes